pkg/sql/create_type.go

```diff
@@ -370,6 +391,58 @@ func CreateEnumTypeDesc(
 	}).BuildCreatedMutableType(), nil
 }
 
+// CreateCompositeTypeDesc creates a new composite type descriptor.
+func CreateCompositeTypeDesc(
+	params runParams,
+	id descpb.ID,
+	compositeTypeList []tree.CompositeTypeElem,
+	dbDesc catalog.DatabaseDescriptor,
+	schema catalog.SchemaDescriptor,
+	typeName *tree.TypeName,
+) (*typedesc.Mutable, error) {
+	// Ensure there are no duplicates in the input enum values.
+	seenLabels := make(map[tree.Name]struct{})
+	labels := make([]string, len(compositeTypeList))
+	types := make([]*types.T, len(compositeTypeList))
+	for i, value := range compositeTypeList {
+		_, ok := seenLabels[value.Label]
+		if ok {
+			return nil, pgerror.Newf(pgcode.InvalidObjectDefinition,
+				"composite type definition contains duplicate label %q", value)
+		}
+		labels[i] = string(value.Label)
+		typ, err := tree.ResolveType(params.ctx, value.Type, params.p.semaCtx.TypeResolver)
+		if err != nil {
+			return nil, err
```

Comment:
Should we wrap the error in something which references the label? Seems like things could get confusing otherwise.

Reply:
I didn't realize that. Fair enough. Let's stay consistent then.